nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -1136,3 +1256,58 @@ ngx_http_modsecurity_drop_action(request_rec *r)
     ctx->r->connection->error = 1;
     return 0;
 }
+
+    static ngx_int_t
+ngx_http_variable_get_modsec_latency(ngx_http_request_t* r,
+    ngx_http_variable_value_t* v, uintptr_t data)
+{
+    return NGX_OK;
+}
+
+    static ngx_int_t
+ngx_http_variable_get_modsec_mode(ngx_http_request_t* r,
+    ngx_http_variable_value_t* v, uintptr_t data)
+{
+    return NGX_OK;
+}
+
+    static ngx_int_t
+ngx_http_modsecurity_set_modsec_latency(ngx_http_request_t *r,
+    ngx_http_variable_value_t *v, ngx_msec_int_t modsec_latency)
+{
+    if (v->data == NULL) {
+        v->data = ngx_pnalloc(r->pool, NGX_TIME_T_LEN + 4);
+        if (v->data == NULL) {
+            return NGX_ERROR;
+        }
+    }
+
+    v->len = ngx_sprintf(v->data, "%T.%03M", (time_t)(modsec_latency) / 1000, modsec_latency % 1000) - v->data;
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+
+    return NGX_OK;
+}
+
+static void
+ngx_http_modsecurity_set_modsec_mode(ngx_http_request_t* r,
+    ngx_http_variable_value_t* v, ngx_uint_t modsec_mode)
+{
+    u_char *p;
+    size_t len;
+    if (modsec_mode == 0) {
```

Comment:
nit: Might be worth to define a macro so that WAF modes are more readable. What do you think?